src/testpool.jl

```diff
@@ -152,15 +153,20 @@ function resize_test_engine_pool(size::Int64, generator::Function = get_next_eng
             engines[new_name] = 0
             TEST_ENGINE_POOL.next_id += 1
         end
-        Threads.@sync for engine in engines
-            if length(engines) > size
-                @info("Deleting engine", engine.first)
-                @async try
-                    delete_engine(get_context(), engine.first)
-                catch
-                    # The engine may not exist if it hasn't been used yet
-                end
-                delete!(engines, engine.first)
+        # Move the first length - size engines to the list of engines to delete
+        engines_to_delete = String[]
+        while length(engines) > size
+            engine_name = first(engines).first
+            push!(engines_to_delete, engine_name)
+            delete!(engines, engine_name)
+        end
+        # Asynchronously delete the engines
+        Threads.@sync for engine in engines_to_delete
+            @info("Deleting engine", engine)
+            @async try
+                delete_engine(get_context(), engine.first)
```

Comment:
An error that was silenced by the try catch. I'll add some error reporting in there as well. I'm sure there will be other warnings that we would like some indication of in the future, rather than having them swallowed.